### PR TITLE
Convert CoffeeScript to JavaScript

### LINKED ISSUE #162: youmightnotneedcoffeescript
Seriously. Javascript is usable now thanks to ES6. We should migrate the coffeescript code to ES6.

### PR DESCRIPTION
**Pull request by nickserv**
Fix #162

### CODE REVIEW

File: src/js/index.js
Comment (Abbondanzo) [inline src/js/index.js:124]: The tiniest of nits: the original coffeescript code generated a closure here and fired the `handleChange` function on load. This will have to be manually recreated or at least invoked once because all elements display on initial load

**Review by Abbondanzo — COMMENTED** [review]
This is fantastic, thanks for doing all this work! I have one small nit but once that's fixed I'll gladly merge this in

File: .prettierrc.json
Comment (jarikmarwede) [inline .prettierrc.json]: Maybe we should add prettier to the `package.json` now. I don't know whether you forgot or just have it installed globally, but I think it would be beneficial to everyone working on this repo.

File: .prettierIgnore
Comment (jarikmarwede) [inline .prettierIgnore]: Why did you ignore `prism.js` but not `prism.css`?

**Review by jarikmarwede — APPROVED** [review]
I have some questions and suggestions concerning prettier, but they are not important enough to block merging.

File: src/js/index.js
Comment (Abbondanzo) [inline src/js/index.js]: This is a constant variable, so it won't allow reassignment. Should be `let` instead
```suggestion
    for (let j = 0, len1 = ref1.length; j < len1; j++) {
```

File: .prettierrc.json
Comment (nickserv) [reply .prettierrc.json]: I use the VSCode extension, but that's a good idea. Done.

**Comment by nickserv** [comment]
I was also thinking of modernizing some other things but I didn't want to make this PR too complex. Any opinions?
1. Convert Stylus to modern CSS (already has nesting and variables, would mostly just be adding brackets like JS)
2. Convert Jade to something more HTML based (like EJS or React/JSX)
3. Update Gulp for performance and support
4. Remove Gulp in favor of a modern bundler (like esbuild, Vite, or swc), native web code, or a static site generator (like Astro)

File: .prettierrc.json
Comment (nickmccurdy) [reply .prettierrc.json:5]: I'm skipping further Prettier tweaks in this PR.

See #263

File: src/js/index.js
Comment (nickserv) [reply src/js/index.js:124]: Thanks, didn't realized that's what `do` was for

**Comment by jarikmarwede** [comment]
> I was also thinking of modernizing some other things but I didn't want to make this PR too complex. Any opinions?
> 
> 1. Convert Stylus to modern CSS (already has nesting and variables, would mostly just be adding brackets like JS)
> 
> 2. Convert Jade to something more HTML based (like EJS or React/JSX)
> 
> 3. Update Gulp for performance and support
> 
> 4. Remove Gulp in favor of a modern bundler (like esbuild, Vite, or swc), native web code, or a static site generator (like Astro)

1. I would also suggest getting rid of Stylus, but as far as I know CSS currently does not support [nesting](https://caniuse.com/css-nesting). The only way to get nesting that I could find was using [this](https://www.npmjs.com/package/postcss-nesting) PostCSS plugin. Maybe that could be an option.
2. That would definitely be nice. Maybe just plain HTML would also be enough, as this site isn't that big and the few dynamic things could be done with some JS.
3. I don't think updating Gulp would be needed if point 4 was choosen
4. I think this would also be a good idea. Personally I would tend to just go native. But we would have to think about how to load the code examples.

Hearing @Abbondanzo's opinion on this would be interesting.

**Comment by nickserv** [comment]
> 1. I would also suggest getting rid of Stylus, but as far as I know CSS currently does not support [nesting](https://caniuse.com/css-nesting). The only way to get nesting that I could find was using [this](https://www.npmjs.com/package/postcss-nesting) PostCSS plugin. Maybe that could be an option.
1. Oops, I thought CSS nesting was implemented, turns out it's just a [working draft](https://drafts.csswg.org/css-nesting/). [Chrome has it planned](https://bugs.chromium.org/p/chromium/issues/detail?id=1095675) though, and we could use that plugin in the interim.

**Comment by Abbondanzo** [comment]
> 1. Convert Stylus to modern CSS (already has nesting and variables, would mostly just be adding brackets like JS)
> 2. Convert Jade to something more HTML based (like EJS or React/JSX)
> 3. Update Gulp for performance and support
> 4. Remove Gulp in favor of a modern bundler (like esbuild, Vite, or swc), native web code, or a static site generator (like Astro)

1. I'm in favor of this, and certainly out of scope for this PR. There are tons of CSS preprocessors out there that are more popular and would have first-class support from the likes of prettier, gulp, etc. I've used SASS plenty in the past and the bracketless syntax is something that grows on ya
2. Jade templates (now called Pug) aren't that far of a departure from HTML. The low JavaScript overhead is what makes this site great. To switch over to React, you'd have to come up with a proper mechanism for loading all of the comparison files. Could they all be turned into components? Probably. It would make it easier to add features that I've been thinking about, like a TS/JS toggle but there's a fair bit of work that would have to go into this one.
3. On board for upgrades. Gulp has only bumped one major version in the nine years since this site was made, but surely in all that time there are more than a handful of improvements
4. If you do decide to switch away to JSX, then this would make sense. I don't have much experience with any of those bundlers or static site generators but if they are quicker and offer more features, then by all means. Gulp is incredibly dependable, loads in a few seconds, and will survive the heat death of the universe

**Review by Abbondanzo — APPROVED** [review]

**Pull request merged**
